microsoftgraph/docs-conceptual/tutorial-grant-delegated-api-permissions.md

```diff
@@ -0,0 +1,166 @@
+```
+
+## Step 2: Create a client service principal
+
+The first step in granting consent is to [create the service principal](/powershell/module/microsoft.graph.applications/new-mgserviceprincipal?view=graph-powershell-1.0&preserve-view=true). To do so, you'll need the `App Id` of your application.
+
+### Register an application with Azure AD
+
+If the application is not available, register an application with Azure AD.
+
+```powershell
+New-MgApplication -DisplayName 'My application' | 
+  Format-List Id, DisplayName, AppId, SignInAudience, PublisherDomain
+```
+
+```Output
+Id              : 40cbfad6-f138-4fb4-9e7f-5a44044efbd6
+DisplayName     : My application
+AppId           : 05210c44-437f-4a40-bd38-b5b4eaf251ef
+SignInAudience  : AzureADandPersonalMicrosoftAccount
+PublisherDomain : Contoso.com
+```
+
+### Create a service principal for the application
+
+```powershell
+New-MgServicePrincipal -AppId '05210c44-437f-4a40-bd38-b5b4eaf251ef' | 
+  Format-List Id, DisplayName, AppId, SignInAudience
+```
+
+```Output
+Id             : 22c1770d-30df-49e7-a763-f39d2ef9b369
+DisplayName    : My application
+AppId          : 05210c44-437f-4a40-bd38-b5b4eaf251ef
+SignInAudience : AzureADandPersonalMicrosoftAccount
+```
+
+## Step 3: Grant delegated permission to the client service principal
+
+To create a delegated permission grant, you'll need the following information:
+
+1. **ClientId** - object Id of the client service principal authorized to act on behalf of the user. In this case, the service principal we created in step 2.
+1. **ConsentType** - `AllPrincipals` to authorize all users in the tenant or `Principal` for a single user.
+1. **PrincipalId** - Id of the user for *Principal* consents, `null` for *AllPrincipals* consents.
+1. **ResourceId** - object Id of the service principal representing the resource app in the tenant.
+1. **Scope** - space-delimited list of permission claim values, for example `User.Read.All`.
+
+In this example, the object Id of the resource service principal is `2cab1707-656d-40cc-8522-3178a184e03d`. You'll grant `Group.Read.All` scope to the service principal and grant consent on behalf of all users in the tenant.
+
+```powershell
+$params = @{
+  "ClientId" = "22c1770d-30df-49e7-a763-f39d2ef9b369"
+  "ConsentType" = "AllPrincipals"
+  "ResourceId" = "2cab1707-656d-40cc-8522-3178a184e03d"
+  "Scope" = "Group.Read.All"
+}
+
+New-MgOauth2PermissionGrant -BodyParameter $params | 
+  Format-List Id, ClientId, ConsentType, ExpiryTime, PrincipalId, ResourceId, Scope
+```
+
+```Output
+Id          : DXfBIt8w50mnY_OdLvmzadDQeqbRp9tKjNm83QyGbTw
+ClientId    : 22c1770d-30df-49e7-a763-f39d2ef9b369
+ConsentType : AllPrincipals
+PrincipalId :
+ResourceId  : a67ad0d0-a7d1-4adb-8cd9-bcdd0c866d3c
+Scope       : Group.Read.All
+```
+
+### Step 4: Assign more or revoke delegated permissions to the service principal
+
+You can add more or reduce scopes to an already existing oauth2PermissionGrant object.
```

Comment:
Maybe a little more direct? "You can add or remove scopes from an existing oauth2PermissionGrant object."